src/System.IO.Compression/src/System/IO/Compression/Crc32Helper.cs

```diff
@@ -450,14 +453,36 @@ internal static class Crc32Helper
             0x264B06E6u
         };
 
+        private static unsafe bool IsZlibCrcAvailable()
+        {
+            try
+            {
+                // Make a P/Invoke into zlib crc32 to ensure we're able to find and use it.
+                // If we are, then use zlib.
+                Interop.zlib.crc32(0, null, 0);
+                return true;
+            }
+            catch
+            {
+                // Otherwise, fallback to managed implementation if zlib isn't available
+                Debug.Write("zlib unavailable");
+                return false;
+            }
+        }
+
         // Calculate CRC based on the old CRC and the new bytes 
         // See RFC1952 for details.
-        static public uint UpdateCrc32(uint crc32, byte[] buffer, int offset, int length)
+        public static uint UpdateCrc32(uint crc32, byte[] buffer, int offset, int length)
         {
             Debug.Assert((buffer != null) && (offset >= 0) && (length >= 0)
                        && (offset <= buffer.Length - length), "check the caller");
 
-            Debug.Assert(BitConverter.IsLittleEndian, "UpdateCrc32 Expects Little Endian");
+            return s_useZlib ? ZlibCrc32(crc32, buffer, offset, length) : ManagedCrc32(crc32, buffer, offset, length); 
+        }
+
+        private static uint ManagedCrc32(uint crc32, byte[] buffer, int offset, int length)
+        {
+            Debug.Assert(BitConverter.IsLittleEndian, "Crc32Managed Expects Little Endian");
```

Comment:
nit: comment should match function name i.e. ""ManagedCrc32 Expects Little Endian"